llvm/lib/IR/ConstantRange.cpp

```diff
@@ -440,6 +440,11 @@ bool ConstantRange::isAllNonNegative() const {
   return !isSignWrappedSet() && Lower.isNonNegative();
 }
 
+bool ConstantRange::isAllPositive() const {
+  // Empty and full set are automatically treated correctly.
+  return !isSignWrappedSet() && Lower.isStrictlyPositive();
+}
```

Comment:
Please split this out into a separate patch and add it to the exhaustive test in `TEST_F(ConstantRangeTest, Negative)`.

Reply:
Thanks, fixing the issues in isAllPositive made me realize that this patch was wrong. Fixed now.